metabase/metabase
# Add tab-key navigation for metadata editor

Adds a custom "tab flow" for dataset metadata editor to make it easier to customize metadata for every column. It should be possible to navigate through every dataset field using the "Tab" key and go through each field's 'essential' properties (display name, description, special/semantic type, and real database column for native datasets).

### To Verify

_I recommend watching the demo video below. In the video, I go through and explain every change. Also, the keystrokes are highlighted there, so it can make it easier to test the change locally._

1. Open any dataset, click its name to open the details sidebar, and click "Customize metadata"
2. Ensure the first column is selected and you see its metadata in the right sidebar. Also, make sure the "Display name" input is focused with all the text selected
3. Hit the "Tab" key a few times. The flow should be: "Display name" > "Description" > "Database column this maps to" (only if you're on a native dataset, the field shouldn't be there for GUI ones) > "Special type" > Next column
4. Once you go through a list of essential fields for a single column and then hit "Tab" while the "Special type" picker is focused, the editor should switch to the next column and "Display name" should become focused
5. Go through a few columns in that way to ensure "tab navigation" works correctly
6. Try navigating backward using "Shift + Tab". You should be able to get back to the previous columns
7. Click on a random column and try using the "tab navigation" again, it shouldn't break
8. Click on the last column and. tab through essential fields. Once you hit "Tab" while being on the "Special type" picker, you should end up on the very first column

### Demo

https://user-images.githubusercontent.com/17258145/146763648-72bbc3b6-312e-42d5-8077-a3dccc21819c.mov
